azurerm/resource_arm_autoscale_setting_test.go

```diff
@@ -0,0 +1,833 @@
+		CheckDestroy: testCheckAzureRMAvailabilitySetDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMAutoscaleSettingExistsfunc,
+					resource.TestCheckResourceAttr(
+						"azurerm_autoscale_setting.test", "enabled", "true"),
+					resource.TestCheckResourceAttr(
+						"azurerm_autoscale_setting.test", "profile.#", "2"),
+					resource.TestCheckResourceAttr(
+						"azurerm_autoscale_setting.test", "profile.0.name", "defaultProfile"),
+					resource.TestCheckResourceAttr(
+						"azurerm_autoscale_setting.test", "profile.0.rule.#", "2"),
+					resource.TestCheckResourceAttr(
+						"azurerm_autoscale_setting.test", "profile.1.fixed_date.#", "1"),
+					resource.TestCheckResourceAttr(
+						"azurerm_autoscale_setting.test", "notification.#", "1"),
+				),
+			},
+		},
+	})
+}
+
+func testCheckAzureRMAutoscaleSettingExistsfunc(s *terraform.State) error {
+	resourceName := "azurerm_autoscale_setting.test"
+	rs, ok := s.RootModule().Resources[resourceName]
+	if !ok {
+		return fmt.Errorf("Not found: %s", resourceName)
+	}
+
+	autoscaleSettingName := rs.Primary.Attributes["name"]
+	resourceGroup, hasResourceGroup := rs.Primary.Attributes["resource_group_name"]
+	if !hasResourceGroup {
+		return fmt.Errorf("Bad: no resource group found in state for Autoscale Setting: %s", autoscaleSettingName)
+	}
+
+	asClient := testAccProvider.Meta().(*ArmClient).autoscaleSettingsClient
+	ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+	resp, err := asClient.Get(ctx, resourceGroup, autoscaleSettingName)
+	if err != nil {
+		if resp.StatusCode == http.StatusNotFound {
+			return fmt.Errorf("Bad: Autoscale Setting %q (resource group: %q) does not exist", autoscaleSettingName, resourceGroup)
+		}
+
+		return fmt.Errorf("Bad: Get on autoscaleSettingsClient: %s", err)
+	}
+
+	return nil
+}
+
+func testCheckAzureRMAutoscaleSettingDestroy(s *terraform.State) error {
+	asClient := testAccProvider.Meta().(*ArmClient).autoscaleSettingsClient
+	ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_autoscale_setting" {
+			continue
+		}
+
+		name := rs.Primary.Attributes["name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+
+		resp, err := asClient.Get(ctx, resourceGroup, name)
+
+		if err != nil {
+			return nil
+		}
+
+		if resp.StatusCode != http.StatusNotFound {
+			return fmt.Errorf("Autoscale Setting still exists:\n%#v", resp.AutoscaleSetting)
+		}
+	}
+
+	return nil
+}
+
+func testAccAzureRMAutoscaleSetting_basic(rInt int) string {
+	return fmt.Sprintf(`
+    variable "time_zone" {
+      type    = "string"
+      default = "Pacific Standard Time"
+    }
+
+    variable "adminPwd" {
+      type    = "string"
+      default = "^l&f_ZwX2L4C"
+    }
+
+    resource "azurerm_resource_group" "test" {
+      name     = "acctas%d"
+      location = "West US"
```

Comment:
Typically we don't hard code location in test cases (because we will pass it as an environment variable). Take [this as an example](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/resource_arm_redis_cache_test.go#L476).